priorityq.py

```diff
@@ -60,10 +61,25 @@ def __init__(self, iterable=()):
         args:
             iterable: an optional iterable to add to the priority queue. Items
                       added this way will be given a priority of None.
+
+        each item inside iterable can be either:
+        * A QNode object
+        * A container with value, priority
+        * A non-iterable value
+
         """
         self.heap = BinaryHeap(iterable=())
         for item in iterable:
-            self.insert(item)
+            try:
+                is_container = len(item) == 2
```

Comment:
Ok, look at the doc string. Iterables have to be packaged using the (val, priority) notation. I think it's reasonable. Let me know what you think.